src/meson.build

```diff
@@ -19,9 +19,16 @@ gdbus_src = custom_target(
 
 daemon_builddir = meson.current_build_dir()
 
+unitdir = ''
+systemd = dependency('systemd', required: false)
+if systemd.found()
+  unitdir = systemd.get_pkgconfig_variable('systemdsystemunitdir')
+endif
+
 conf = configuration_data()
 conf.set_quoted('GETTEXT_PACKAGE', 'ubuntu-advantage-desktop-daemon')
 conf.set_quoted('PROJECT_VERSION', meson.project_version())
+conf.set('libexecdir', libexecdir)
```

Comment:
I would make a separate conf object just for the `systemd_unit` - that way we can't accidentally make a collision with unexpected defines / substitutions.